게시판 / 게시글 임시 컴포넌트 ,페이지 구조 구현

booungyi opened the pull request:
댓글 / 대댓글 컴포넌트 , 페이지 구조구현
상세설명은 notion 작업 -> 프론트 1차 설계

booungyi committed:
게시판 / 게시글 임시 컴포넌트 ,페이지 구조 구현
댓글 / 대댓글 컴포넌트 , 페이지 구조구현

yulmucha commented on src/components/posts/post.service.ts:
components 폴더는 주로 UI 컴포넌트를 모아두는 곳이라, 현재 파일에 담긴 API 호출이나 데이터 처리 로직은 별도의 서비스 레이어에 두는 편이 더 깔끔해요.
아래처럼 수정해 보시면 어떨까요?
- 파일 이동 경로: src/components/… -> src/services/…
- 파일 이름: posts.ts

이렇게 분리해 두면 UI 코드와 비즈니스 로직이 명확히 구분되어 코드 구조가 한눈에 들어오고 유지보수나 확장 시에도 훨씬 수월해집니다.

yulmucha commented on src/components/posts/likedPost/page.tsx:
지금 이 `page.tsx` 파일은 `src/components/` 폴더에 두셨는데요, Next.js의 라우팅 구조를 올바르게 활용하려면 `app/` 디렉토리 안에 위치시켜 주시는 게 좋아요. 이렇게 해 두면:
- 라우트 자동 등록이 정상적으로 이루어져요.
- 페이지별 레이아웃이나 서버 컴포넌트 설정이 의도한 대로 작동해요.
- 코드를 구조적으로 구분해 두면, 다른 개발자(또는 본인)가 나중에 유지·보수할 때도 한눈에 이해하기 쉬워요.

예를 들어:
📁 project-root
 ┗ 📂 src
   ┣ 📂 app
   ┃ ┗ 📂 posts
   ┃   ┗ 📂 likedPost
   ┃     ┗ page.tsx    ← 여기에 두기
   ┗ 📂 components    ← UI 컴포넌트만!
     ┗ 📂 posts
       ┗ 📂 likedPost
         ┗ (UI 컴포넌트 파일들)

이렇게 분리해 보시면 어떨까요?

yulmucha commented on src/pages/boards/page.tsx:
```diff
@@ -0,0 +1,14 @@
+    cache: 'no-store',
+  });
+  const data = await res.json();
+
```
여기서 직접 `fetch`를 사용하고 계신데요,
`src/pages/boards/[boardId]/posts/page.tsx` 에서는 `fetchPosts()` 같은 추상화된 함수를 활용하셨던 것처럼,
여기에서도 동일한 패턴을 적용해 보면 어떨까요?

장점:
- 코드 일관성 유지
- 공통 에러 처리·로깅 로직 재사용
- 나중에 API 호출 방식을 변경할 때 한 곳만 수정하면 OK

heojae93 replied:
아~ 새로만들어서 커밋하고 푸시했습니다 ^ㅇ^ ~ 乃

yulmucha commented on src/pages/posts/[postId]/page.tsx:
```diff
@@ -0,0 +1,100 @@
+  params: { postId: string };
+}
+
+export default function PostDetailPage({ params }: Props) {
```
지금 `src/pages/posts/[postId]/page.tsx` 파일은 최상단에 `'use client'`가 선언되어 있어서 페이지 전체가 클라이언트 컴포넌트로 동작해요. 이렇게 되면
- 서버 사이드 렌더링(SSR)과 초기 로딩 속도, SEO 이점을 활용하기 어렵고
- 불필요한 클라이언트 번들 사이즈가 커질 수 있어요.

Next.js 15의 App Router에서는 기본적으로 서버 컴포넌트로 작성하고,
오직 아래와 같은 클라이언트 전용 기능만 별도의 클라이언트 컴포넌트로 분리하는 걸 추천드려요.
- useState, useEffect 같은 React 훅
- useRouter 등의 클라이언트 전용 훅
- 사용자 인터랙션(좋아요 토글, 수정, 삭제) 처리

이 구조로 바꿔보시면 더 깔끔하고 확장하기도 쉬워질 거예요. 화이팅입니다! 🚀

yulmucha commented on src/pages/posts/new/page.tsx:
```diff
@@ -0,0 +1,44 @@
+import { API_URL } from '@/env/constants';
+import { useRouter } from 'next/navigation';
+
+export default function CreatePostPage() {
```
이 페이지도 `src/pages/posts/[postId]/page.tsx`와 마찬가지로 기본은 서버 컴포넌트로 작성하시고,
`useState`, `useRouter`, 버튼 클릭 핸들러처럼 클라이언트 기능이 필요한 부분만 별도의 클라이언트 컴포넌트로 분리해 보시면 좋습니다.

예를 들어:
1. 서버 컴포넌트 (page.tsx):
    - 데이터 초기화
    - 레이아웃, 마크업 렌더링
2. 클라이언트 컴포넌트 (CreatePostForm.tsx):
    - useState, useRouter
    - handleCreate 같은 사용자 인터랙션 처리

이런 방식으로 관심사(Concerns)를 분리하시면 유지보수성과 확장성 모두 한층 좋아집니다.

booungyi committed:
경로 수정
-> /app/boards , /app/posts  ,
[postId]/page.tsx->
서버 컴포넌트 , 유저컴포넌트 분리

booungyi committed:
posts/new/page.tsx-> 서버 클라이언트
CreatePostForm.tsx->  use client
서버 , 유저 분리

lch0208 committed:
게시판 서버 컴포넌트, 유저 컴포넌트 분리

booungyi committed:
boards/[boardId]/posts/page.tsx
-> try catch 추가
mockService.ts 테스트용 서비스 추가
components/boards/BoardList.tsx href={`/src/app/boards/${board.id}/posts`}->href={`/boards/${board.id}/posts`} 변경

booungyi committed:
boards/[boardId]/posts/page.tsx
-> try catch 추가
mockService.ts 테스트용 서비스 추가
components/boards/BoardList.tsx href={`/src/app/boards/${board.id}/posts`}->href={`/boards/${board.id}/posts`} 변경

lch0208 committed:
게시판 목록조회 구조구현 및 경로수정

booungyi committed:
Merge remote-tracking branch 'origin/fornt_boardPostComment' into fornt_boardPostComment

# Conflicts:
#	src/components/boards/BoardList.tsx

booungyi committed:
Merge remote-tracking branch 'origin/fornt_boardPostComment' into fornt_boardPostComment

# Conflicts:
#	src/components/boards/BoardList.tsx

booungyi committed:
Merge remote-tracking branch 'origin/fornt_boardPostComment' into fornt_boardPostComment
